src/main/java/com/io/image/manager/controller/ImageController.java

```diff
@@ -83,14 +90,43 @@ public ImageController(ImageService imageService, PrometheusMeterRegistry mr, Ap
 
         if (image.isPresent()) {
             byte[] imageArray = imageService.dumpImage(image.get(), conversionInfo);
+            logRequest(filename, props.getOriginServer(), true, operations);
             outboundTrafficSummary.record(imageArray.length);
             return ResponseEntity.ok(imageArray);
         }
+        logRequest(filename, props.getOriginServer(), false, operations);
 
         throw new ResponseStatusException(HttpStatus.NOT_FOUND);
     }
 
     private OriginServer originFromHost(String host) {
         return new OriginServer(String.format("https://%s/", host));
     }
+
+    private void logRequest(String filename, String origin, boolean isPresent, List<ImageOperation> operations){
+        try {
+            if (writer == null){
+                writer = new BufferedWriter(new FileWriter(logPath, true));
```

Comment:
Are you sure there won't be any race conditions? I don't know spring boot well and I have no idea if controllers are singletons or not, @Przemyslaw5 do you happen to know how it works?

Reply:
I'm not sure